### PR TITLE
Added experimentalFragmentVariables compatibility.

### PR DESCRIPTION
**Pull request by lucasconstantino**
I'm experimenting with fragments on [react-apollo-fragments](http://github.com/lucasconstantino/react-apollo-fragments) and have reached a point to deal with fragment variables. The folks at the core of *graphql-js* project have added some experimental parsing for this, and I would love to use it in my own library. But as I use `graphql-tag`, as as this library does not export the full parser API below it, I'm decided to pull-request.

You can better understand what I'm looking for here https://github.com/graphql/graphql-js/blob/master/src/language/parser.js#L99-L115

I tried to find a workaround, but I found no way other than replace *graphql-tag* usage.

- [x] feature
- [ ] blocking
- [ ] docs

### CODE REVIEW

**Comment by jnwng** [comment]
thanks @lucasconstantino — i didnt know there were runtime flags for the graphql parser, it seems like something that we can patternize to make it easier to flag all of these. 

one question i have before i merge: what's the use case for enabling / disabling this flag on the fly? was that mainly done for testing purposes? (just trying to see if that's generally a pattern for actual usage)

File: src/index.js [hunk 1]
```diff
@@ -118,14 +118,15 @@ function stripLoc(doc, removeLocAtThisLevel) {
...
+var experimentalFragmentVariables = false;
 function parseDocument(doc) {
   var cacheKey = normalize(doc);
 
   if (docCache[cacheKey]) {
     return docCache[cacheKey];
   }
 
-  var parsed = parse(doc);
+  var parsed = parse(doc, { experimentalFragmentVariables });
```
Comment (jnwng) [inline src/index.js]: unfortunately we're not compiling this file just yet, so im going to make this an explicit object

**Comment by lucasconstantino** [comment]
@jnwng I'm currently working on making a first-level component Fragment for react-apollo. It's a work in progress, and pretty much a proposition to react-apollo itself, but it will rely on parameterized fragments such as:

```gql
fragment NamedFragment ($arg: String!) on Type {
  field (arg: $arg)
}
```

You can see the work in progress here: https://github.com/lucasconstantino/react-apollo-fragments

**Comment by lucasconstantino** [comment]
Just for reference: there is a long discussion happening on the community since 2016 here: https://github.com/facebook/graphql/issues/204

**Comment by erickreutz** [comment]
Can someone help me better understand how to use these fragment variables? The documentation is pretty sparse and doesn't elude to how to use the feature entirely. 

I'm under the assumption that I can do something like the following:

```
query UserInfo {
  currentUser {
    id
    firstName
    ...UserAvatar(size: 100)
  }
}
fragment UserAvatar(size: Int!) on User {
  avatarUrl(size: $size)
}
```

I do this and I get the following error:
> Syntax Error: Expected Name, found (

Am I missing something?

**Comment by axelhzf** [comment]
@erickreutz do you know how to make it work? Do I need to make any change on the backend to understand this kind of queries? I am getting this error from the backend `GraphQL error: Syntax Error: Expected "on", found (`

**Comment by jimmycob** [comment]
Hello,

I have same error than @erickreutz 
Am I missing something?

Thanks

**Pull request merged**